service/src/stock_take/update.rs

```
@@ -40,19 +27,56 @@ pub struct UpdateStockTakeInput {
     pub status: Option<StockTakeStatus>,
 }
 
+#[derive(Debug, PartialEq)]
 pub enum UpdateStockTakeError {
     DatabaseError(RepositoryError),
     InternalError(String),
     InvalidStore,
     StockTakeDoesNotExist,
     CannotEditFinalised,
+    /// Holds list of affected stock take line ids
+    LinesNotCounted(Vec<String>),
+    /// Holds list of affected stock take line ids
+    SnapshotCountCurrentCountMismatch(Vec<String>),
+}
+
+/// returns uncounted stock take lines
+fn check_stock_take_lines_counted(stock_take_lines: &Vec<StockTakeLine>) -> Option<Vec<String>> {
+    let uncounted: Vec<String> = stock_take_lines
+        .into_iter()
+        .filter(|line| line.line.counted_number_of_packs.is_none())
+        .map(|line| line.line.id.clone())
+        .collect();
+    if uncounted.len() > 0 {
+        return Some(uncounted);
+    }
+    None
+}
+
+fn check_snapshot_matches_current_count(
+    stock_take_lines: &Vec<StockTakeLine>,
+) -> Option<Vec<String>> {
```

i think this method should return StockTakeLine. In API we would want to return an error with array of stock take lines (connector), this makes it very easy for front end to action on the error. Returning just ids would mean we need to make another loader call in API or f/e would need to require the stock take with lines an filter out the ids

---

Makes sense but how is FE suppose to solve the conflict. Should it ask the user to recount?

I return the stock take line id and the stock line now.

348a6cf

---

>  Should it ask the user to recount?

Yeah no idea how they would solve them, but giving them the info would allow them all the options.